### PR TITLE
Refactor HTML Template data parsing

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Refactor HTML Template data parsing

Get presentation in controller; Write URL in parser

## Motivation and Context
Adhere to existing Viewer structure. Simplify writing tests.

## How Has This Been Tested?
Validated changes in the local environment.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
At this point I think the structure of initializing the HTML Template is finalized. Next, I will try to write tests for HTMLTemplateController.

Then I will remove the messaging functionality from HTMLTemplateRenderer. I think that is the final piece of refactoring for the main PR.

@ezequielc please review. Thanks!

File: src-js/Viewer/Renderer/HTMLTemplateRenderer.js [hunk 1]
```diff
@@ -15,79 +17,85 @@ service.render = (scheduleItem, htmlName, containerName) => {
     return;
   }
 
-  var url = scheduleItem.objectReference;
-  url += url.indexOf("?") > -1 ? "&" : "?";
-  url += "frameElementId=" + "iFrame_" + htmlName;
+  const isStaging = false;
+  const version = isStaging ? "staging" : "stable";
```
Comment (alex-deaconu) [inline src-js/Viewer/Renderer/HTMLTemplateRenderer.js:21]: @ezequielc wondering if we should make this conditional; for preview show `beta`, for everything else (shared templates and eventually displays) show `stable`? Thanks.

File: src-js/Viewer/Renderer/HTMLTemplateRenderer.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 const EntryPoint = require("../EntryPoint");
+const ViewerDataInfo = require("../Data/ViewerDataInfo");
 const Logger = require("../Logger");
 
 const service = {};
 const HTML_TEMPLATE_DOMAIN = "https://widgets.risevision.com";
+const HTML_TEMPLATE_URL = "https://widgets.risevision.com/STAGE/templates/PCODE/src/template.html?presentationId=PID&frameElementId=FRAMEID";
 
-service.render = (scheduleItem, htmlName, containerName) => {
+service.render = (scheduleItem, presentation, htmlName, containerName) => {
```
Comment (ezequielc) [inline src-js/Viewer/Renderer/HTMLTemplateRenderer.js:9]: @alex-deaconu Do we need to pass scheduleItem? Looks like it is only used to get the `objectReference`, which is the same as `presentation.id`, right? Thanks

File: src-js/Viewer/Renderer/HTMLTemplateRenderer.js (hunk 1 above)
Comment (ezequielc) [reply src-js/Viewer/Renderer/HTMLTemplateRenderer.js:21]: @alex-deaconu I am not sure we would like to show `beta` for production Preview. If coming form Apps Beta that makes sense, but I would avoid mixing that. Thanks

File: src-js/Viewer/Renderer/HTMLTemplateRenderer.js (hunk 2 above)
Comment (alex-deaconu) [reply src-js/Viewer/Renderer/HTMLTemplateRenderer.js:9]: @ezequielc yes, you're right. I just kept it for consistency with the other controller. Will remove. Thanks!

File: src-js/Viewer/Renderer/HTMLTemplateRenderer.js (hunk 1 above)
Comment (alex-deaconu) [reply src-js/Viewer/Renderer/HTMLTemplateRenderer.js:21]: @ezequielc we talked. I'll use staging/stable based on config environment. Will update in a separate PR. Thanks.

File: src-js/Viewer/Renderer/HTMLTemplateRenderer.js (hunk 2 above)
Comment (alex-deaconu) [reply src-js/Viewer/Renderer/HTMLTemplateRenderer.js]: @ezequielc updated, thanks!

**Comment by ezequielc** [comment]
@alex-deaconu Looks good 👍 Thanks for the updates

**Pull request merged**